src/webAppCreator.ts

```diff
@@ -519,10 +521,30 @@ class ShellScriptStep extends WebAppCreatorStepBase {
             .replace('%PLAN_SKU%', plan.sku.name)
             .replace('%SITE_NAME%', site.name)
             .replace('%RUNTIME%', site.siteConfig.linuxFxVersion);
-        vscode.workspace.openTextDocument({
-            content: script,
-            language: 'shellscript'
-        }).then(doc => vscode.window.showTextDocument(doc));
+
+        let uri: vscode.Uri;
+        if (vscode.workspace.rootPath) {
+            let count = 0;
+            const maxCount = 1024;
+            
+            while (count < maxCount) {
+                uri = vscode.Uri.file(path.join(vscode.workspace.rootPath, `deploy-${site.name}${count === 0 ? '' : count.toString()}.sh`));
+                if (!vscode.workspace.textDocuments.find(doc => doc.uri.fsPath === uri.fsPath) && !fs.existsSync(uri.fsPath)) {
+                    uri = uri.with({ scheme: 'untitled' });
+                    break;
+                }
+                count++;
+            }
+        }
+
+        if (uri) {
+            const doc = await vscode.workspace.openTextDocument(uri);
+            const editor = await vscode.window.showTextDocument(doc);
+            await editor.edit(editorBuilder => editorBuilder.insert(new vscode.Position(0,0), script));
+        } else {
+            const doc = await vscode.workspace.openTextDocument({ content: script, language: 'shellscript' });
```

Comment:
@PrashanthCorp can you do the same thing for https://github.com/Microsoft/vscode-cosmosdb/issues/1

The language is 'mongo'. The current behavior in the cosmos repo throws an error when I try to save, but this works